src/main/java/com/demo/classroom/Controller/AuthController.java

```diff
@@ -0,0 +1,39 @@
+package com.demo.classroom.Controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.demo.classroom.DTO.RegistrationDTO;
+import com.demo.classroom.Service.RegistrationService;
+import com.demo.classroom.Utility.ErrorMessages;
+
+import jakarta.validation.Valid;
+
+@RestController
+@RequestMapping("/req")
+public class AuthController {
+    @Autowired
+    private RegistrationService registrationService;
+
+    @PostMapping(value = "/signup", consumes = "application/json")
+    public ResponseEntity<String> registerUser(@Valid @RequestBody RegistrationDTO request, BindingResult result) {
+
+        if (result.hasErrors()) {
+            String errorMessage = ErrorMessages.constructErrorMessage(result);
+            return ResponseEntity.badRequest().body(errorMessage.toString());
+        }
+
+        String response = registrationService.registerUser(request);
+        if (response.contains("successfully")) {
+             return ResponseEntity.status(HttpStatus.CREATED).body(response);
+        } else {
+            return ResponseEntity.badRequest().body(response);
+        }
```

Comment:
Avoid using string matching for success or failure checks. Instead, you could modify the RegistrationService to return an object with a status code or a boolean indicating success, along with a message.

```
RegistrationResponse response = registrationService.registerUser(request);
if (response.isSuccess()) {
    return ResponseEntity.status(HttpStatus.CREATED).body(response.getMessage());
} else {
    return ResponseEntity.badRequest().body(response.getMessage());
}

```

The RegistrationResponse class could look like this:
```
public class RegistrationResponse {
    private boolean success;
    private String message;

    // Constructor, getters, setters
}

```

Reply:
done